tamr_unify_client/dataset/uri.py

```diff
@@ -0,0 +1,43 @@
+class DatasetURI:
+    """
+    Indentifier of a dataset.
+
+    :param client: Queried dataset's client.
+    :type client: :class:`~tamr_unify_client.client.Client`
+    :param uri: Queried datset's dataset ID.
+    :type uri: :py:class:`str`
+    """
+
+    def __init__(self, client, uri):
+        self.client = client
+        self._uri = uri
+
+    @property
+    def resource_id(self):
+        """:type: str"""
+        return self._uri.split("/")[-1]
+
+    @property
+    def relative_id(self):
+        """:type: str"""
+        return "datasets/" + self.resource_id
+
+    @property
+    def uri(self):
+        """:type: str"""
+        return self._uri
+
+    def dataset(self):
+        """Dataset that indentifier points to.
```

Comment:
This needs a verb, e.g. "Fetch the dataset that this identifier points to." to make it clear that this takes an action with the server.